### PR TITLE
fix: whirlpool no longer blocks collar of subordination

### LINKED ISSUE #4948: Whirlpool and Collar
As the current ruling, whirlpool still move creature with collar to opponent battlefield but right now it prevents EDAI from chaning side.
Image

### PR DESCRIPTION
**Pull request by dmikalova**
Fixes #4948

Whirlpool was blocking the Collar of Subordination's control-change effect at end of turn. This removes the unintended interference so creatures with Collar correctly change control as expected.

Also adds a skipped test for the Whirlpool + Collar interaction to document the known edge case with Animating Force/Deanimator.

---

> [!NOTE]
> **Medium Risk**
> Changes global effect-resolution precedence by making `mostRecentEffect` always return the latest effect, which can subtly alter many card interactions that relied on persistent-effect prioritization. Runtime impact is confined to effect ordering, but the surface area is broad across gameplay rules.
> 
> **Overview**
> **Effect precedence is updated to “newest wins.”** `GameObject.mostRecentEffect()` no longer prioritizes `persistentEffect` durations and now always applies the most recently-added effect, aligning control/type overrides (e.g. Whirlpool vs. Collar, Animating Force vs. De-Animator) with last-applied behavior.
> 
> **Tests and tooling are adjusted accordingly.** Adds a Whirlpool+Collar regression test, updates the De-Animator/Animating Force expectation to match the new precedence, and hardens vitest assertions/serialization to treat `Card`/`Player`/`Game` objects as identity-only to avoid slow/heap-blowing deep diffs.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit 24585b57d1d6537d656c6e39b85bd17c7b3b6694. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: test/helpers/integrationhelper.js [hunk 1]
```diff
@@ -255,6 +255,74 @@ const customMatchers = {
 // Register custom matchers with vitest's expect.extend
 expect.extend(customMatchers);
```
Comment (dmikalova) [inline test/helpers/integrationhelper.js:257]: When writing these tests and a few others I was getting OOM errors that were irrelevant to the tests, and were just vitest serializing the deepEquals outputs. This flattens that serialization to prevent OOM.

File: test/server/cards/08-AS/DeAnimator.spec.js [hunk 2]
```diff
@@ -201,37 +201,36 @@ describe('De-Animator', function () {
             });
         });
```
Comment (dmikalova) [inline test/server/cards/08-AS/DeAnimator.spec.js:203]: After talking more with Gee and Andrew it looks like this was the wrong take - newest ability is now always considered to override previous ones, including persistent/pervasive abilities like deanimator.

**Review by strib — APPROVED** [review]
